File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cs
public class Solution {
    public int MaxVowels(string s, int k) {
        bool IsVowel(char el){
            if(el == 'a') return true;
            if(el == 'e') return true;
            if(el == 'i') return true;
            if(el == 'o') return true;
            if(el == 'u') return true;
            return false;
        }

        int count = 0;

        for(int i = 0; i < k; i++) if(IsVowel(s[i])) count++;

        int temp = count;

        for(int j = k; j < s.Length; j++)
        {
            if(IsVowel(s[j])) temp++;
            if(IsVowel(s[j - k])) temp--;
            if(temp > count) count = temp;
        }
        return count;
    }
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.ts
function maxVowels(s: string, k: number): number {
    function isVowel(chr: string): boolean {
        if (chr === 'a') return true;
        if (chr === 'e') return true;
        if (chr === 'i') return true;
        if (chr === 'o') return true;
        if (chr === 'u') return true;
        return false;
    }

    let count: number = 0;

    for (let i = 0; i < k; i++) {
        if (isVowel(s[i])) count++;
    }

    let temp: number = count;

    for (let j = k; j < s.length; j++) {
        if (isVowel(s[j])) temp++;
        if (isVowel(s[j - k])) temp--;
        if (temp > count) count = temp;
    }

    return count;
};